### ISSUE TITLE
Ruby 3.0 Upgrade

### ISSUE THREAD

**Issue by bkroeker** [prograils/lit#160]
LIT does not work with Ruby 3.0.  The main issue seems to be method definitions that use `options = {}`, which should now be `**options`.

As a temporary stopgap, I've monkeypatched several LIT methods to use the new method definition in our ruby 3.0 app.

eg. 

```ruby
module LitFrontendHelper
  # identical to Lit::FrontendHelper::TranslationKeyWrapper#translate, except for method definition
  def translate(key, **options)
    count = options[:count]
    options = options.with_indifferent_access
    key = scope_key_by_partial(key)
    key = pluralized_key(key, count) if count
    content = super(key, **options.symbolize_keys)
    content = get_translateable_span(key, content) if !options[:skip_lit] && lit_authorized?
    content
  end
end
```

**Comment by pnikrat** [comment]
We've just tested Lit in fresh app that has Ruby 3.0 and Rails 6.0.3. It works pretty well, no errors or deprecations.

Maybe you've encountered a different problem - zeitwerk module autoloading error? We've solved it in https://github.com/prograils/lit/pull/165

What you've described in this issue is not really a problem. Have you checked [Ruby positional and keyword args](https://www.ruby-lang.org/en/news/2019/12/12/separation-of-positional-and-keyword-arguments-in-ruby-3-0/) ?

According to this:
```
Note that Ruby 3.0 doesn’t behave differently when calling a method which doesn’t accept keyword arguments with keyword arguments. For instance, the following case is not going to be deprecated and will keep working in Ruby 3.0. The keyword arguments are still treated as a positional Hash argument.

def foo(kwargs = {})
  kwargs
end

foo(k: 1) #=> {:k=>1}
```

**Comment by bkroeker** [comment]
Thanks for the response @pnikrat.

I assume you tested the frontend helper stuff in ruby 3?  That's where the issues were that I needed to fix.

The difference between the lit code and the snippet you copied from the "Ruby positional and keyword args" article is that the lit code has a positional argument, `key`, as you can see from the code snippet in my initial post.  That style of method definition is no longer supported.

I had a number of similar method definitions that I had to update in my own code when I upgraded to Ruby 3.  As stated at the top of that article, they showed as deprecations in ruby 2.7 and completely broke in ruby 3.

**Comment by mlitwiniuk** [comment]
@bkroeker You were right, this commit should fix the issue: 6e5a6e335036abc470fbece445393df01b18f4eb, we'll merge it to master (and release new version) later this week

**Comment by mlitwiniuk** [comment]
merged to master. New release next week
